docs/cluster-agent/HORIZONTAL_POD_AUTOSCALING.md

```diff
@@ -0,0 +1,262 @@
+
+```
+
+### Register the External Metrics Provider
+
+Once the DCA is up and running, you can register it as an External Metrics Provider, via the service exposing the port 443.
+
+To do so, simply apply the following RBAC rules:
+
+`kubectl apply -f manifest/hpa-example/rbac-hpa.yaml`
+
+```
+clusterrolebinding.rbac.authorization.k8s.io "system:auth-delegator" created
+rolebinding.rbac.authorization.k8s.io "dca" created
+apiservice.apiregistration.k8s.io "v1beta1.external.metrics.k8s.io" created
+clusterrole.rbac.authorization.k8s.io "external-metrics-reader" created
+clusterrolebinding.rbac.authorization.k8s.io "external-metrics-reader" created
+```
+ 
+Once you have the DCA running and the service registered, you can create an HPA manifest and let the DCA pull metrics from Datadog.
+The following part of the walkthrough explains how you can set up your agents in order to collect metrics from the applications running on your cluster.
+Those metrics will then be available for you to autoscale the resources of your cluster, via the DCA.
+If your agents are already instrumented and configured to communicate with the DCA, you can directly jump to the [running the hpa]() section.
+
+
+## Running the HPA
+
+At this point, you should be seeing:
+```
+PODS
+
+NAMESPACE     NAME                                     READY     STATUS    RESTARTS   AGE
+default       datadog-agent-4c5pp                      1/1       Running   0          14m
+default       datadog-agent-ww2da                      1/1       Running   0          14m
+default       datadog-agent-2qqd3                      1/1       Running   0          14m
+[...]
+default       datadog-cluster-agent-7b7f6d5547-cmdtc   1/1       Running   0          16m
+default       nginx-6757dd8769-5xzp2                   1/1       Running   0          3m
+
+```
+
+Now is time to create a Horizontal Pod Autoscaler manifest. If you take a look at /manifest/cluster-agent/hpa-example/hpa-manifest.yaml, you will see:
+- The HPA is configured to autoscale the Deployment called nginx
+- The maximum number of replicas created will be 5 and the minumum is 1
+- The metric used is `nginx.net.request_per_s` and the scope is `kube_container_name: nginx`. Note that this metric format corresponds to the Datadog one.
+
+Every 30 seconds (this can be configured) Kubernetes will query the DCA to get the value of this metric and will autoscale proportionally if necessary.
+For advanced use cases, it is possible to have several metrics in the same HPA, as you can see [here](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#support-for-multiple-metrics) the largest of the proposed value will be the one chosen.
+
+Now, let's create our nginx deployment:
+
+`kubectl apply -f manifests/cluster-agent/hpa-example/nginx.yaml`
+
+Then, let's apply the HPA
+
+`kubectl apply -f manifests/cluster-agent/hpa-example/hpa-manifest.yaml`
+
+### Stressing your service
+
+Now is time to see the magic happening!
```

Comment:
This sentence can be deceptive, the magic is happening only if everything went well before :)